controllers.py

```diff
@@ -27,3 +28,27 @@ def sl_update(self, states, optimal_actions):
         loss.backward()
         self.policy.optimizer.step()
         return loss.item()
+
+class SLController():
+    def __init__(self):
+        self.policy = torch.load('models/pretrained_vit_onehot_mad.pt')
+        self.optimizer = torch.optim.Adam(self.policy.parameters(), lr=1e-3)
+        self.criterion = torch.nn.MSELoss()
+        self.device = 'cpu'
+
+    def deterministic_forward(self, emg_window):
+        emg_window_tensor = emg_window.unsqueeze(0).to(self.device)
+        emg_window_tensor.swapaxes_(2, 3)
```

Comment:
is this necessary? maybe it is

Reply:
maybe we can restructure the network to have the right axes, or change the source to provide the right inputs to avoid either non-continous memory views or copying vectors around